src/HttpClientInterception/HttpRequestInterceptionBuilder.cs

```diff
@@ -97,6 +97,46 @@ public HttpRequestInterceptionBuilder For(Predicate<HttpRequestMessage>? predica
         return this;
     }
 
+    /// <summary>
+    /// Configures the builder to match any request that meets all the criteria defined by the specified predicates.
+    /// </summary>
+    /// <param name="predicates">
+    /// Delegates to methods which must all return <see langword="true"/> for the
+    /// request to be considered a match.
+    /// </param>
+    /// <returns>
+    /// The current <see cref="HttpRequestInterceptionBuilder"/>.
+    /// </returns>
+    /// <remarks>
+    /// Pass a value of <see langword="null"/> to remove a previously-registered custom request matching predicate.
+    /// </remarks>
+    public HttpRequestInterceptionBuilder For(params Predicate<HttpRequestMessage>[] predicates)
```

Comment:
I wonder if for both of these we shouldn't use `params` and should instead use an `IEnumerable<T>`/`ICollection<T>` as otherwise it's possible to write this code, which is valid C#, but looks weird, and then ultimately gets you an exception at runtime.

```csharp
builder.For(); // Implicit empty array
```

Reply:
The trade-off is the API usage might feel a little clunkier, so we go from this:
```
.For(IsHttpGetForJustEatGitHubOrg, _ => requestCount < 2)
```

To this:
```
.For(new Predicate<HttpRequestMessage>[]
{
   IsHttpGetForJustEatGitHubOrg,
   _ => requestCount < 2,
})
```

An alternative would be something like 
```
.For(IsHttpGetForJustEatGitHubOrg)
.For(_ => requestCount < 2)
```

And have .For(), internally manage a merged collection when `.Build()` is called. What do you think?

Reply:
I don't think chaining the calls in this case is intuitive behaviour, and would make the internals quite complicated to work.

Can this not work?

```csharp
.For(new[] { IsHttpGetForJustEatGitHubOrg, _ => requestCount < 2 })
```

Reply:
I tried it locally, I'm getting a `CS1503` compilation error

```
  Examples.cs(719, 18): [CS1503] Argument 1: cannot convert from 'System.Func<System.Net.Http.HttpRequestMessage, bool>[]' to 'System.Predicate<System.Net.Http.HttpRequestMessage>?'
```

Reply:
> I don't think chaining the calls in this case is intuitive behaviour, and would make the internals quite complicated to work.

Would you suggest I keep the API as is and accept `.For()` will only error at runtime, or make the change to replace `params` with `ICollection<T>`?

Reply:
Personally I'd drop the `params` as it leads the user away from writing code that doesn't work in the first place. It only seems to make sense to me to support params when there can be any number of parameters - the API here doesn't allow that from the validation, so I think it's not an appropriate use case for it, even if it's "nicer" in the expected usage.

_If_ we then wanted to add it at a later date, it would be doable without being a breaking change.